lottie/src/main/java/com/airbnb/lottie/animation/content/ContentGroup.java

```diff
@@ -175,24 +178,40 @@ Matrix getTransformationMatrix() {
     }
 
     // Apply off-screen rendering only when needed in order to improve rendering performance.
-    boolean isRenderingWithOffScreen = lottieDrawable.isApplyingOpacityToLayersEnabled() && hasTwoOrMoreDrawableContent() && layerAlpha != 255;
+    boolean isRenderingWithOffScreen =
+        (lottieDrawable.isApplyingOpacityToLayersEnabled() && hasTwoOrMoreDrawableContent() && layerAlpha != 255) ||
+        (shadowToApply != null && lottieDrawable.isApplyingShadowToLayersEnabled() && hasTwoOrMoreDrawableContent());
+    int childAlpha = isRenderingWithOffScreen ? 255 : layerAlpha;
+
+    Canvas contentCanvas = canvas;
     if (isRenderingWithOffScreen) {
       offScreenRectF.set(0, 0, 0, 0);
-      getBounds(offScreenRectF, matrix, true);
-      offScreenPaint.setAlpha(layerAlpha);
-      Utils.saveLayerCompat(canvas, offScreenRectF, offScreenPaint);
+      getBounds(offScreenRectF, parentMatrix, true);
```

Comment:
This one got lost in the diff, but is very important - `matrix` was being passed here, causing incorrect transformations being applied when `applyOpacitiesToLayers` is true; see the `precomp_opacity_killer.json` testcase.